src/create-reminder-signup/lambda/lambda.ts

```diff
@@ -20,8 +31,14 @@ export const handler = (
 
 	const signup = body as OneOffSignup;
 
-	return Promise.resolve({
-		statusCode: 200,
-		body: JSON.stringify(signup),
-	});
+	return identityAccessTokenPromise
+		.then((token) => getIdentityIdByEmail(signup.email, token))
+		.then((identityId) => ({
+			statusCode: 200,
+			body: JSON.stringify({ ...signup, identityId }),
+		}))
+		.catch((err) => ({
+			statusCode: 400,
+			body: JSON.stringify(err.message),
+		}));
```

Comment:
This probably isn't the right thing to do. This should be a 400 when identity can't find the email, but if identity has a different error, or maybe we're having an error in sending it to identity, it should be a 500?

Reply:
Yes, and the logic will be different for pre-contribution signups where we create an account if identity returns 404.

But focusing for now on post-contribution (where we expect the user to have an identityId) - you're right that anything besides 404 (from identity) should be a 500 for the client.
In practice the client UI will probably show a generic error to the user, regardless of the status code.